be/src/exprs/agg/distinct.h

```diff
@@ -486,72 +497,36 @@ class DictMergeAggregateFunction final
                 }
             }
         } else {
-            const auto& binary_column = down_cast<const BinaryColumn&>(*columns[0]);
+            const auto& binary_column = down_cast<const BinaryColumn&>(*data_column);
             agg_state.update(mem_pool, binary_column.get_slice(row_num));
         }
 
-        agg_state.over_limit = agg_state.set.size() > DICT_DECODE_MAX_SIZE;
-    }
-
-    void update_batch_single_state(FunctionContext* ctx, size_t chunk_size, const Column** columns,
```

Comment:
Why remove `update_batch_single_state` ?

Reply:
not we add dict_merge into aggregate_resolver with add_general_mapping_notnull so there will not be a NullableAggregateFunctionUnary  above it, just use `AggregateFunctionBatchHelper`, so update_batch_single_state has been dealed in `AggregateFunctionBatchHelper`  by calling `update`